### PR TITLE
Fix KeyError: 'checkin_times'

### LINKED ISSUE #4465: KeyError: 'checkin_times'
https://sentry.eventyay.com/eventyay/dev-api/issues/102/

```
KeyError: 'checkin_times'
(9 additional frame(s) were not displayed)
...
  File "app/api/helpers/permissions.py", line 43, in decorator
    return fn(*args, **kwargs)
  File "app/api/attendees.py", line 125, in before_update_object
    if obj.checkin_times and data['checkin_times'] not in obj.checkin_times.split(","):

KeyError: 'checkin_times'
```

### PR DESCRIPTION
**Pull request by ananyo2012**
Fixes #4465

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `nextgen` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Fixes KeyError 'checkin_times' when `data['is_checked_in'] = false`

### CODE REVIEW

**Comment by ananyo2012** [comment]
Hey @subham-padia I have solved the issue. But kept in WIP as I need some more information. What if a the status of an attendee is changes from `is_checked_in = true` to `is_checked_in=false` Should we ideally remove the `checkin_times` data or keep it ? I have removed the data in this case but let me know if we need to keep it.

File: app/api/attendees.py [hunk 1]
```diff
@@ -118,12 +118,15 @@ def before_update_object(self, obj, data, kwargs):
             raise ForbiddenException({'source': 'User'}, 'You are not authorized to access this.')
 
         if 'is_checked_in' in data:
-            if data['is_checked_in'] and 'checkin_times' not in data:
-                raise UnprocessableEntity({'pointer': '/data/attributes/checkin_times'},
-                                          "Check in time missing while trying to check in attendee")
-
-            if obj.checkin_times and data['checkin_times'] not in obj.checkin_times.split(","):
-                data['checkin_times'] = '{},{}'.format(obj.checkin_times, data['checkin_times'])
+            if data['is_checked_in']:
```
Comment (rahuldecoded) [inline app/api/attendees.py]: If in `data` `checkin_times` doesn't present they it will definitely throw KeyError. Try to handle this situation.

Comment (ananyo2012) [reply app/api/attendees.py]: [This line](https://github.com/fossasia/open-event-orga-server/pull/4489/files#diff-5e5ae0e99c073cd15ba3cd3e8bbf7c72R122) checks if `checkin_times` is present and handles the error

**Comment by ananyo2012** [comment]
@shubham-padia @niranjan94 Can have a look at my [previous comment](https://github.com/fossasia/open-event-orga-server/pull/4489#issuecomment-331659552) ?

**Comment by shubham-padia** [comment]
I think the data should be kept. cc: @magdalenesuo

**Comment by ananyo2012** [comment]
@shubham-padia I am assuming we need to keep the data as per your previous comment. This PR is ready for review now.

**Comment by ananyo2012** [comment]
Explanation of the fix : If `data['is_checked_in']` is true it takes the 'checkin_times' data for the attendee and throws error if it isn't present. If `data['is_checked_in']` is false we do nothing as ideally the attendee is not checked in and we don't need to save the 'checkin_times' data.

**Comment by ananyo2012** [comment]
@niranjan94 @shubham-padia Can you have a look if anymore changes are required?

File: app/api/attendees.py (hunk 1 above)
Comment (poush) [reply app/api/attendees.py]: This can easily go in a single single, like 
```
if 'is_checked_in' in data and data['is_checked_in']:
```

**Comment by poush** [comment]
Added a single comment above. Apart from that lgtm!

**Comment by ananyo2012** [comment]
@poush Made the requested changes

**Pull request merged**